src/ptwt/conv_transform_2.py

```diff
@@ -214,51 +212,52 @@ def wavedec2(
     if level is None:
         level = pywt.dwtn_max_level([data.shape[-1], data.shape[-2]], wavelet)
 
-    result_lst: List[
-        Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor, torch.Tensor]]
-    ] = []
+    result_lst: list[tuple[torch.Tensor, torch.Tensor, torch.Tensor]] = []
     res_ll = data
     for _ in range(level):
         res_ll = _fwt_pad2(res_ll, wavelet, mode=mode)
         res = torch.nn.functional.conv2d(res_ll, dec_filt, stride=2)
         res_ll, res_lh, res_hl, res_hh = torch.split(res, 1, 1)
         to_append = (res_lh.squeeze(1), res_hl.squeeze(1), res_hh.squeeze(1))
         result_lst.append(to_append)
-    result_lst.append(res_ll.squeeze(1))
+
     result_lst.reverse()
+    res_ll = res_ll.squeeze(1)
+    result: WaveletCoeffDetailTuple2d = res_ll, *result_lst
```

Comment:
might be an opportunity to rethink this data structure to have more nested parts rather than a variable length part

Reply:
I agree that such a data structure would preferable.
However, the current data structure is chosen to follow the conventions of `pywt` (which expects a tuple or list of the current form, see [the API reference](https://pywavelets.readthedocs.io/en/latest/ref/2d-dwt-and-idwt.html#pywt.waverec2)). Also, moving to more nested parts would probably break significantly more user code.

Reply:
in that case, better to stick to what's already here, but just leave a comment on the definition of the data structure that this is the reason :)